### PR TITLE
Use std::unique_ptr instead of std::shared_ptr for the boundary tractions.

### PR DESCRIPTION
**Pull request by bangerth**
Part of #2375 and #2811.

### CODE REVIEW

**Comment by bangerth** [comment]
The `traction_ascii_data` test fails. I'll have to investigate what is happening, so ignore this PR for the moment.

File: source/boundary_traction/ascii_data.cc [hunk 1]
```diff
@@ -35,14 +35,11 @@ namespace aspect
     void
     AsciiData<dim>::initialize ()
     {
-      const std::map<types::boundary_id,std::shared_ptr<BoundaryTraction::Interface<dim> > >
-      bvs = this->get_boundary_traction();
-      for (typename std::map<types::boundary_id,std::shared_ptr<BoundaryTraction::Interface<dim> > >::const_iterator
-           p = bvs.begin();
-           p != bvs.end(); ++p)
+      std::set<types::boundary_id> traction_bi;
```
Comment (tjhei) [inline source/boundary_traction/ascii_data.cc]: you introduced a local set here that you are filling but not using.

Comment (bangerth) [reply source/boundary_traction/ascii_data.cc]: Yes, good point. Stupid copy-paste mistake.

**Comment by bangerth** [comment]
That did the trick, thanks @tjhei for the pointer!

**Pull request merged**